armi/reactor/blueprints/__init__.py

```diff
@@ -304,10 +304,20 @@ def _prepConstruction(self, cs):
             self._assembliesBySpecifier.clear()
             self.assemblies.clear()
 
+            # retrieve current count of assemblies to restore after
+            # creating blueprints assemblies. This is particularly useful for
+            # doing snapshot based runs with database loads where the
+            # assembly counter is likely not equal to 0.
```

Comment:
Add "To ensure that the assembly count is not a function of how many times load state/snapshots has been called"